src/game/file.c

```diff
@@ -227,6 +227,7 @@ static void initialize_saved_game(void)
 
     sound_city_init();
 
+    building_construction_clear_type();
```

Comment:
Why do we need this?

Reply:
Because android keeps some things in memory on exit, if that code isn't there the last used construction will still be active upon restarting the game.

Reply:
Isn't that also a problem when loading scenarios then?

Reply:
If I remember well this code works for both cities and scenarios.

Reply:
Only for campaign scenarios (which are basically saved games from mission1.pak), custom scenarios don't call this function when starting.

Reply:
Yeah, I knew it did work but it was a while ago that I fixed it so I didn't really remember it too well. Thanks for checking 👍